include/targets/generic/reduction_kernel_host.h

```diff
@@ -11,7 +11,7 @@ namespace quda
     Functor<Arg> t(arg);
 
     reduce_t value = t.init();
-
+#pragma omp parallel for collapse(2) reduction(Functor <Arg>::apply : value)
```

Comment:
Is this legal?  I thought for reduction one would need either one of the predefined reductions (+,-,*,&,|,^,&& and II), or it has to be declared as a custom reduction? A custom reduction here would be awkward because we'd need a name for the new reduction, which would vary with whatever Arg::apply is? 
Or is this cool as per the standard?

Reply:
This method is standard: writing this made me realize why @jcosborn @jxy wanted to define the reduction operators as static methods for the SYCL/OMP targets, which is required for this to work.

Reply:
[c52fac3](https://github.com/lattice/quda/pull/1479/commits/c52fac3e6e75dc3479d0c01935e714cc04c91802)